### PR TITLE
Package repository UI adapted based on repository permissions API

### LINKED ISSUE #5542: Unable to edit or delete newly created flux package repository
Scenario:
* user A has cluster admin access to namespace A
* user A logs in to portal, switch context to namespace A and creates a new flux repo there
* the repo is created successfully but EDIT, REFRESH and DELETE button are disabled. A bit ironic since REFRESH button was enabled in previous screen

Detailed repro steps:
See attached .txt file, some of steps 1-22, having to do with setting up ns A and user A 

screen 1
Screen Shot 2022-10-20 at 9 12 39 PM

screen 2
Screen Shot 2022-10-20 at 9 05 37 PM

[multi-tenant-setup.txt](https://github.com/vmware-tanzu/kubeapps/files/9835624/multi-tenant-setup.txt)

### PR DESCRIPTION
**Pull request by castelblanque**
### Description of the change

This PR sits on top of PR #5604 and adapts UI to use the API endpoint introduced to return allowed permissions for package repositories.
It also fixes #5542, opened for Flux.
Areas affected are:
- `Add package repository` button is enabled only if the user has `create` permission at current namespace or cluster level.
- `Edit/Delete` buttons on each repo are enabled if user has `update` permission in repository type.
- `Show repositories in all namespaces` switch is enabled if user has `list` permissions cluster-wide.

### Benefits

Real permissions granted to the current user are checked to adapt the UI when managing package repositories.

### Possible drawbacks

Some more fine-tuning of the UI is needed to take advantage of the new permissions API endpoint.

### Applicable issues

- fixes #5542

### CODE REVIEW

File: dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx [hunk 1]
```diff
@@ -99,56 +102,30 @@ function PkgRepoList() {
...
-    });
+    const rbac = new Map<string, PackageRepositoriesPermissions>();
+    reposPermissions.forEach(p => rbac.set(JSON.stringify(p.plugin), p));
+    setReposRBAC(rbac);
 
     // Cluster-wide check
-    Kube.canI(cluster, "kubeapps.com", "apprepositories", "list", "")
-      .then(allowed => setCanSetAllNS(allowed))
-      ?.catch(() => setCanSetAllNS(false));
-  }, [
```
Comment (antgamdia) [inline dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx:132]: Really happy to finally be removing this kind of logic!

File: dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx [hunk 2]
```diff
@@ -99,56 +102,30 @@ function PkgRepoList() {
...
-    return plugin ? reposRBAC.get(JSON.stringify(plugin))?.namespaced.update || false : false;
+    if (!plugin) {
+      return false;
+    }
+    return reposRBAC.get(JSON.stringify(plugin))?.namespace["update"] || false;
   };
 
   const canAddRepos = () => {
-    return [...reposRBAC.values()].some(r => r.global?.create || r.namespaced.create);
+    return [...reposRBAC.values()].some(r => r.global["create"] || r.namespace["create"]);
```
Comment (antgamdia) [inline dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx:128]: Perhaps not for this PR, but wondering whether we should move the `"create"`, `"update"`, etc literals to an enum.

**Review by antgamdia — APPROVED** [review]
Awesome, thanks for performing the change end-to-end!

File: dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx (hunk 2 above)
Comment (beni0888) [reply dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx:128]: Agree 🙂

Comment (castelblanque) [reply dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx:128]: Yep, totally agree.

**Pull request merged**